flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/expressions/TemporalTypesTest.scala

```diff
@@ -1122,6 +1154,90 @@ class TemporalTypesTest extends ExpressionTestBase {
 
   }
 
+  @Test
+  def testToTimestampLtzShanghai(): Unit = {
```

Comment:
1. Please also test the minimum and maximum values for each numeric type this function supports. (I think we might need some boundary protection for timestamp because we can only support 9999-12-31)
2. Add a test case with NULL input
3. Add some tests about unsupported input types
4. Add some tests that exceeds the precision of timestamp, such as `TO_TIMESTAMP_LTZ(0.01, 3)`

Reply:
nice comments